crates/reth/evm/src/utils.rs

```diff
@@ -28,23 +28,31 @@ pub fn wei_to_sats(wei: U256) -> (U256, U256) {
 /// # Note
 ///
 /// A [`Descriptor`], if invalid does not create an [`WithdrawalIntent`].
-pub fn collect_withdrawal_intents(
-    receipts: impl Iterator<Item = Option<Receipt>>,
-) -> impl Iterator<Item = WithdrawalIntent> {
-    receipts
-        .flatten()
-        .flat_map(|receipt| receipt.logs)
-        .filter(|log| log.address == BRIDGEOUT_ADDRESS)
-        .filter_map(|log| {
-            WithdrawalIntentEvent::decode_log(&log, true)
+pub fn collect_withdrawal_intents<'a, I>(
+    tx_receipt_pairs: I,
+) -> impl Iterator<Item = WithdrawalIntent> + 'a
+where
+    I: Iterator<Item = (&'a TransactionSigned, &'a Receipt)> + 'a,
```

Comment:
Need to update the docstring.

Also might consider creating a type alias for that nasty tuple as well.